### ISSUE TITLE
[BUG] Kanister Controller cannot update Status on Kubernetes versions < v1.20

### ISSUE THREAD

**Issue by pavannd1** [kanisterio/kanister#1576]
**Describe the bug**
Kanister Controller fails to add `Status` to ActionSets on Kubernetes Clusters of version < v1.20. 

**To Reproduce**
1. Build controller image from `master` branch
2. Install it on a cluster running K8s version 1.19 or lower
3. Create a Blueprint and an ActionSet to execute actions
4. Kanister controller fails to update the status
```
  Warning  Update Failed  5s    Kanister Controller  Could not update ActionSet: ActionSet.cr.kanister.io "hookbp-wmwr2" is invalid: status.progress.lastTransitionTime: Invalid value: "null": status.progress.lastTransitionTime in body must be of type string: "null"
```

**Expected behavior**
ActionSet should successfully run all the actions.

**Screenshots**
Example Blueprint:
```
apiVersion: cr.kanister.io/v1alpha1
kind: Blueprint
metadata:
  name: hook-blueprint
  namespace: kanister
actions:
  post-export:
    kind: Namespace
    phases:
    - args:
        command:
        - /bin/sh
        - -c
        - |
          echo "This is a test phase"
        image: alpine:latest
        podOverride:
          serviceAccountName: myrelease-kanister-operator
      func: KubeTask
      name: hookPhase
```

Example ActionSet:
```
apiVersion: cr.kanister.io/v1alpha1
kind: ActionSet
metadata:
  generateName: hookbp-
  namespace: kanister
spec:
  actions:
  - name: post-export
    blueprint: hook-blueprint
    object:
      kind: Namespace
      name: default
      namespace: default
```

Describe ActionSet after creation:
```
Name:         hookbp-wmwr2
Namespace:    kanister
Labels:       <none>
Annotations:  <none>
API Version:  cr.kanister.io/v1alpha1
Kind:         ActionSet
Metadata:
  Creation Timestamp:  2022-07-29T22:46:40Z
  Generate Name:       hookbp-
  Generation:          1
  Managed Fields:
    API Version:  cr.kanister.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:generateName:
      f:spec:
        .:
        f:actions:
    Manager:         kubectl-create
    Operation:       Update
    Time:            2022-07-29T22:46:40Z
  Resource Version:  57845
  Self Link:         /apis/cr.kanister.io/v1alpha1/namespaces/kanister/actionsets/hookbp-wmwr2
  UID:               71839463-5a08-4090-a9e7-d19724c018b6
Spec:
  Actions:
    Blueprint:  hook-blueprint
    Name:       post-export
    Object:
      Kind:       Namespace
      Name:       default
      Namespace:  default
Events:
  Type     Reason         Age   From                 Message
  ----     ------         ----  ----                 -------
  Warning  Update Failed  5s    Kanister Controller  Could not update ActionSet: ActionSet.cr.kanister.io "hookbp-wmwr2" is invalid: status.progress.lastTransitionTime: Invalid value: "null": status.progress.lastTransitionTime in body must be of type string: "null"
```

No issues on clusters with K8s v1.20 or above
```
Name:         hookbp-dzg8v
Namespace:    kanister
Labels:       <none>
Annotations:  <none>
API Version:  cr.kanister.io/v1alpha1
Kind:         ActionSet
Metadata:
  Creation Timestamp:  2022-07-29T20:19:07Z
  Generate Name:       hookbp-
  Generation:          9
  Managed Fields:
    API Version:  cr.kanister.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:generateName:
      f:spec:
    Manager:      kubectl-create
    Operation:    Update
    Time:         2022-07-29T20:19:07Z
    API Version:  cr.kanister.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:spec:
        f:actions:
      f:status:
        .:
        f:actions:
        f:error:
          .:
          f:message:
        f:progress:
          .:
          f:lastTransitionTime:
          f:percentCompleted:
        f:state:
    Manager:         controller
    Operation:       Update
    Time:            2022-07-29T20:19:09Z
  Resource Version:  12830367
  UID:               521f4c46-4fda-4ace-9f9b-523aaec33988
Spec:
  Actions:
    Blueprint:  hook-blueprint
    Name:       post-export
    Object:
      API Version:
      Group:
      Kind:             Namespace
      Name:             default
      Namespace:        default
      Resource:
    Preferred Version:
Status:
  Actions:
    Blueprint:  hook-blueprint
    Defer Phase:
      Name:
      State:
    Name:     post-export
    Object:
      API Version:
      Group:
      Kind:         Namespace
      Name:         default
      Namespace:    default
      Resource:
    Phases:
      Name:   hookPhase
      State:  complete
  Error:
    Message:
  Progress:
    Last Transition Time:  2022-07-29T20:19:15Z
    Percent Completed:     100.00
  State:                   complete
Events:
  Type    Reason           Age    From                 Message
  ----    ------           ----   ----                 -------
  Normal  Started Action   3m52s  Kanister Controller  Executing action post-export
  Normal  Started Phase    3m52s  Kanister Controller  Executing phase hookPhase
  Normal  Ended Phase      3m45s  Kanister Controller  Completed phase hookPhase
  Normal  Update Complete  3m45s  Kanister Controller  Updated ActionSet 'hookbp-dzg8v' Status->complete
  Normal  Update Complete  3m44s  Kanister Controller  Updated ActionSet 'hookbp-dzg8v' Status->complete
```

**Environment**
Kubernetes Version/Provider: 1.18.2/AWS (deployed using KOPS)
Storage Provider: EFS
Cluster Size (#nodes): 3
Data Size: N/A

**Comment by pavannd1** [comment]
Seems to be related to https://github.com/kubernetes/kubernetes/issues/86811
This issue was fixed in Kubernetes v1.20 - https://github.com/kubernetes/kubernetes/pull/95423
To support earlier versions, the workaround seems to be to use a pointer (*metav1.Time) instead of `metav1.Time`.

As downstream users of Kanister still use older versions, the controller will require temporary changes to successfully work on Kubernetes versions < v.1.20.

**Comment by pavannd1** [comment]
@viveksinghggits I was trying out some fixes last week but couldn't make much progress. Could you please take a look?
